includes/class-wc-payments-explicit-price-formatter.php

```diff
@@ -106,21 +109,38 @@ public static function unregister_formatted_woocommerce_price_filter() {
 	 * @return string
 	 */
 	public static function get_explicit_price( string $price, WC_Abstract_Order $order = null ) {
-		if ( false === static::should_output_explicit_price() ) {
-			return $price;
-		}
 		if ( null === $order ) {
 			$currency_code = get_woocommerce_currency();
 		} else {
 			$currency_code = $order->get_currency();
 		}
 
-		if ( ! empty( $currency_code ) ) {
-			$price_to_check = html_entity_decode( wp_strip_all_tags( $price ) );
+		return static::render_explicit_price_with_currency( $price, $currency_code );
+	}
+
+	/**
+	 * Returns a rendered price value suffixed with the appropriate currency code (if necessary).
+	 *
+	 * In multicurrency stores, order and price values are rendered with currency suffix.
+	 * This method only renders the currency suffix if appropriate (see `should_output_explicit_price`).
+	 *
+	 * @param string  $price A price value (as a string).
+	 * @param ?string $currency_code Currency of the price.
+	 *
+	 * @return string Price value with currency code suffix if necessary.
+	 */
+	public static function render_explicit_price_with_currency( string $price, ?string $currency_code ) {
```

Comment:
Thought: consider renaming this from `render_explicit_price_with_currency` to `get_explicit_price_with_currency`, since this does not actually render the string to the page.

Reply:
Fair enough, I tend to use `render` for generating html, but `get` is more consistent with this class/codebase.

Reply:
Updated in 5260756